xla/stream_executor/gpu/gpu_stream.cc

```diff
@@ -26,6 +26,9 @@ limitations under the License.
 #include "xla/stream_executor/platform.h"
 #include "xla/stream_executor/stream.h"
 
+#if GOOGLE_CUDA
```

Comment:
why not use nvtx_utils.h?

Reply:
The first argument of `nvtxNameCuStreamA` is a `CUstream`: https://nvidia.github.io/NVTX/doxygen/group___r_e_s_o_u_r_c_e___n_a_m_i_n_g.html#gacada16c62463b268f8ede952ec77f766, which is available in this context as a `GpuStreamHandle`.

Doing this in nvtx_utils.h would I believe either require promoting some kind of abstract stream handle into TSL, or resorting to `void*`/similar; I thought the current approach was pragmatic.

Reply:
The advantage would be to have one file that mediates the interaction with NVTX.

nvtx_utils.h already forward declares structs like ProfilerDomain, can we do something similar for CUstream?

Reply:
I believe this is a little different, in that the concrete meaning of `ProfilerDomain` is opaque everywhere outside nvtx_utils. Here there is a `CUstream` that comes from inside XLA's GPU abstraction layers that needs to be routed to NVTX.

Note that it is impractical to ensure `GOOGLE_CUDA` is defined in all situations that `nvtx_utils.h` is included. That reality was part of the motivation for the introduction of `ProfilerDomain` in https://github.com/openxla/xla/pull/9896.

I pushed a separate commit trading type safety for NVTX confinement, but it's just glorified `void*`. Let me know if you prefer this.

Reply:
Resolving based on https://github.com/openxla/xla/pull/13603#discussion_r1644594297